docs/source/index.rst

```diff
@@ -40,12 +40,10 @@ The Jupyter notebook
 
 .. toctree::
    :maxdepth: 1
-   :caption: Developer Documentation
+   :caption: Contributor Documentation
 
-   development_intro
```

Comment:
What about `index.html#contributor-documentation` or `contributing.html` as a landing place?

> contributors (which I define broadly and beyond strictly development activities)

`CONTRIBUTING.rst` is not strictly limited to anything and can be filled with as broad information as necessary.

> [...] translation, outreach materials creation [...]

As far as that's specific to the Notebook, it should IMHO go into `CONTRIBUTING.rst`, everything beyond that should probably go to http://jupyter.readthedocs.org/, right?

Commenting stuff out is IMHO about the worst possible thing to do (I'm shivering just thinking about it), I'm certainly not going to do that.